### PR TITLE
add for-user-id into header

### PR DESCRIPTION
**Pull request by ngoclamvt123**

### CODE REVIEW

**Review by rikyhidayat21 — APPROVED** [review]

**Review by VincentWijaya — APPROVED** [review]

File: lib/xendit_api/api/virtual_account.rb [hunk 1]
```diff
@@ -10,8 +10,9 @@ def create(params)
         params[:is_closed] = true if params[:is_closed].nil?
         params[:is_single_use] = true if params[:is_single_use].nil?
         params[:expected_amount] = params[:amount] unless params[:amount].nil?
-
-        response = client.post(PATH, params)
+        headers = {}
+        headers['for-user-id'] = params[:for_user_id] unless params[:for_user_id].nil?
```
Comment (rslhdyt) [inline lib/xendit_api/api/virtual_account.rb:14]: sorry for the late comment, I think the wrapper should not be responsible for handling the headers condition.
just pass the headers to the post method, and let the client handle the value of the header.

Comment (rslhdyt) [reply lib/xendit_api/api/virtual_account.rb:14]: and also the source of headers should not come from the params

File: lib/xendit_api/client.rb [hunk 2]
```diff
@@ -78,8 +78,11 @@ def get(url, params = nil)
       response.body
     end
 
-    def post(url, params)
-      response = @connection.post(url, params)
+    def post(url, params, headers = nil)
```
Comment (rslhdyt) [inline lib/xendit_api/client.rb:81]: I think this changes is enough @ngoclamvt123 to handle custom headers.

File: lib/xendit_api/api/virtual_account.rb [hunk 3]
```diff
@@ -6,12 +6,12 @@ module Api
...
       PATH = '/callback_virtual_accounts'.freeze
 
-      def create(params)
+      def create(params, headers = nil)
         params[:is_closed] = true if params[:is_closed].nil?
         params[:is_single_use] = true if params[:is_single_use].nil?
         params[:expected_amount] = params[:amount] unless params[:amount].nil?
 
-        response = client.post(PATH, params)
+        response = client.post(PATH, params, headers)
```
Comment (okfriansyah-moh) [inline lib/xendit_api/api/virtual_account.rb:14]: If you already specified headers = nil , does this headers also nil in here? Or it depends on the function that uses it? If the function in mekari payment use this and add headers with value, Does the headers in this also will using those value?

Comment (rslhdyt) [reply lib/xendit_api/api/virtual_account.rb:14]: I think we can set empty hash for default value of headers

Comment (okfriansyah-moh) [reply lib/xendit_api/api/virtual_account.rb:14]: headers = {} ya you mean sal

**Review by okfriansyah-moh — APPROVED** [review]

File: lib/xendit_api/api/virtual_account.rb (hunk 3 above)
Comment (rslhdyt) [reply lib/xendit_api/api/virtual_account.rb:14]: yes @okfriansyah-moh

**Review by rslhdyt — APPROVED** [review]

**Pull request merged**